test/LogTest.php

```diff
@@ -0,0 +1,222 @@
+        $opusLog->addFilter($filter);
+
+        $infoMessage = 'Info level Message';
+        $opusLog->info($infoMessage);
+
+        $opusLog->setLevel(\Zend_Log::NOTICE);
+        $noticeMessage = 'Notice Level Message';
+        $opusLog->notice($noticeMessage);
+
+        $opusLog->setLevel(\Zend_Log::ERR);
+        $errorMessage = 'Error level Message';
+        $opusLog->err($errorMessage);
+
+        $content = $this->readLog();
+
+        $this->assertNotContains($infoMessage, $content);
+        $this->assertNotContains($noticeMessage, $content);
+        $this->assertContains($errorMessage, $content);
+    }
+
+    public function testSetLevelNegativeLevel()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $opusLog->setLevel(-1);
+    }
+
+    public function testSetLevelStringArgument()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $exceptionMessage = 'Level should be of Integer type and cannot be negative';
+        $this->setExpectedException(\InvalidArgumentException::class, $exceptionMessage);
+
+        $opusLog->setLevel('TestLevel');
+    }
+
+    public function testSetLevelStringLevel()
+    {
+        $opusLog = $this->getOpusLog();
+
+        $opusLog->setLevel('7');
+
+        $this->assertEquals(\Zend_Log::DEBUG, $opusLog->getLevel());
+    }
+
+    public function testGetLevel()
+    {
+        $opusLog = $this->getOpusLog();
+        $level = $opusLog->getLevel();
+
+        $this->assertEquals(\Zend_Log::INFO, $level);
+    }
+
+    public function testGetLevelReturnsNull()
+    {
+        $opusLog = $this->getOpusLog();
+        $opusLog->setLevel(null);
+        $level = $opusLog->getLevel();
+
+        $this->assertNull($level);
+    }
+
+    public function testGet()
+    {
+        \Zend_Registry::set('Zend_Log', $this->getOpusLog());
+        $log = Log::get();
+
+        $this->assertSame(\Zend_Registry::get('Zend_Log'), $log);
+    }
+
+    public function testDrop()
+    {
+        $log = $this->getOpusLog();
+        \Zend_Registry::set('Zend_Log', $log);
+
+        $logger1 = Log::get();
+        Log::drop();
+        $logger2 = Log::get();
+
+        $this->assertNotSame($logger1, $logger2);
+    }
+
+    protected function getOpusLog()
+    {
+        $format = '%priorityName%: %message%' . PHP_EOL;
+        $formatter = new \Zend_Log_Formatter_Simple($format);
+
+        $this->logFile = fopen('php://temp', 'rw');
+        $writer = new \Zend_Log_Writer_Stream($this->logFile);
+        $writer->setFormatter($formatter);
+
+        $logger = new Log($writer);
+
+        $level = \Zend_Log::INFO;
+        $logger->setLevel($level);
+
+        return $logger;
+    }
+
+    protected function readLog()
+    {
+        rewind($this->logFile);
+        $content = '';
+        while (true) {
```

Comment:
This can probably be simplified.

Reply:
Any thoughts here? You can write `while ($string = fgets($this->logFile)) {` and safe yourself the `if`. That is common and much more readable.